webrecorder/browsertrix
# Allow users to change password

(https://github.com/ikreymer/browsertrix-cloud/issues/21) Users can change their own password from their account settings page.

### Testing
Pre-req: run `yarn` to install new dependencies
1. Start server with `yarn start-dev`
2. Log in
3. Go to http://localhost:9870/account/settings
4. Click "Change password" and try to change your password. Verify that new password/confirmation works, and that an incorrect password will show an error

### Screenshots

Account settings page:
<img width="913" alt="Screen Shot 2021-11-23 at 1 24 45 PM" src="https://user-images.githubusercontent.com/4672952/143131755-0905d672-5fc1-4e3b-a820-95ef5d042d63.png">

Form:
<img width="413" alt="Screen Shot 2021-11-23 at 1 25 37 PM" src="https://user-images.githubusercontent.com/4672952/143131773-0148732b-5b5a-4fc9-845d-b5ebdc2b0da7.png">

On password confirmation mismatch:
<img width="416" alt="Screen Shot 2021-11-23 at 1 25 03 PM" src="https://user-images.githubusercontent.com/4672952/143131809-a7c22950-0406-4db2-a109-61100ac410cd.png">

On wrong password:
<img width="413" alt="Screen Shot 2021-11-23 at 1 27 03 PM" src="https://user-images.githubusercontent.com/4672952/143131909-5d41b00d-f981-40d4-b1ce-194df7f27eaf.png">

On success:
<img width="929" alt="Screen Shot 2021-11-23 at 1 25 33 PM" src="https://user-images.githubusercontent.com/4672952/143131935-4c495cd0-72c5-4d56-879b-56101c626e68.png">

### Opinions

Uses an [xstate finite state machine](https://github.com/statelyai/xstate/tree/main/packages/xstate-fsm) for predictive handling of form states.
